R/downloadableReactTable.R

```diff
@@ -0,0 +1,229 @@
+#'  shinyApp(ui = fluidPage(fluidRow(column(width = 12,
+#'    downloadableReactTableUI("object_id1",
+#'                        downloadtypes = c("csv", "tsv"),
+#'                        hovertext     = "Download the data here!",
+#'                        contentHeight = "300px",
+#'                        singleSelect  = FALSE)))),
+#'    server = function(input, output) {
+#'      mydataRowIds <- function(){
+#'        rownames(head(mtcars))[c(2, 5)]
+#'      }
+#'      selectedrows <- downloadableReactTable(
+#'        id               = "object_id1",
+#'        logger           = "",
+#'        filenameroot     = "mydownload1",
+#'        downloaddatafxns = list(csv = reactiveVal(mtcars), tsv = reactiveVal(mtcars)),
+#'        tabledata        = reactiveVal(mtcars),
+#'        selection        = mydataRowIds,
+#'        table_options    = list(rownames = TRUE,
+#'                             caption  = "This is a great table!"))
+#'      observeEvent(selectedrows(), {
+#'        print(selectedrows())
+#'      })})
+#'}
+#'
+#' @export
+downloadableReactTableUI <- function(id,
+                                downloadtypes = NULL,
+                                hovertext     = NULL,
+                                contentHeight = "200px",
+                                singleSelect  = FALSE) {
+    ns <- shiny::NS(id)
+    list(
+        shiny::conditionalPanel(
+            condition = "output.displayButton",
+            ns        = ns,
+            shiny::span(
+                id    = ns("reactTableButtonDiv"),
+                class = "periscope-downloadable-table-button",
+                style = ifelse(length(downloadtypes) > 0, "", "display:none"),
+                downloadFileButton(ns("reactTableButtonID"),
+                                   downloadtypes,
+                                   hovertext))),
+        reactable::reactableOutput(ns("reactTableOutputID")),
+        shiny::tags$input(
+            id    = ns("reactTableOutputHeight"),
+            type  = "text",
+            class = "shiny-input-container hidden",
+            value = contentHeight),
+        shiny::tags$input(
+            id    = ns("reactTableSingleSelect"),
+            type  = "text",
+            class = "shiny-input-container hidden",
+            value = singleSelect)
+        )
+}
+
+
+#' downloadableReactTable module server function
+#'
+#' Server-side function for the downloadableReactTableUI.  This is a custom
+#' high-functionality table paired with a linked downloadFile
+#' button.
+#'
+#' downloadFile button will be hidden if \code{downloadableReactTable} parameter \code{downloaddatafxn} or
+#' \code{downloadableReactTableUI} parameter \code{downloadtypes} is empty
+#'
+#' Generated table can highly customized using function \code{?DT::datatable} same arguments
+#'  except for `options` and `selection` parameters.
+#'
+#' For `options` user can pass the same \code{?DT::datatable} options using the same names and
```

Comment:
Again, I can tell this is copied/pasted, you refer to DT here.  This table should work VERY differently from DT, so it can't be matched like this copy/pasted.

Reply:
updated